style: adjust title padding on explorer sidebar

Sync271 opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1482

### Description:
<!-- Add a description of the changes proposed in the pull request. -->

### Preview:
<!-- Insert your gif/screenshot/loom recording here -->

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

ZeRego commented:
Looks like there is a formatting error. I suspect it is the indentation of the `<p>` element. We have a pre-commit hook that should be linting and formatting the code 🤔  Have you ran `yarn install` in your machine ? 
These commands should help find the issues:
```
yarn lint
yarn format
```

The text and button don't look well aligned, might need a bit more work.
![Screenshot 2022-03-02 at 17 19 15](https://user-images.githubusercontent.com/9117144/156413973-5d064768-1406-4c41-81d4-2679abce2dd8.png)

Sync271 commented:
> The text and button don't look well aligned, might need a bit more work.

@ZeRego Yup, I reverted all my changes and just reworked padding for the title, which looks better now.
Removed padding on all sides except the top and reduced padding for the top.

ZeRego commented:
![Screenshot 2022-03-03 at 09 15 20](https://user-images.githubusercontent.com/9117144/156533960-e778ceff-f4ff-4bcb-8000-d5704fdba7c4.png)
Why it is showing the 1 and 2 inside the bar ? @owlas
